File: Data Structures & Algorithms/missing-number/submission-2.py
class Solution:
    def missingNumber(self, nums: List[int]) -> int:
        ledger = 0
        for num in nums:
            ledger |= 1 << num

        
        ledger ^= 2**(len(nums)+1) - 1

        return ledger.bit_length() - 1
        

File: Data Structures & Algorithms/largest-number/submission-2.cpp
class Solution {
public:
    string largestNumber(vector<int>& nums) {
        vector<string> strs;
        strs.reserve(nums.size());

        for (int num : nums) {
            strs.push_back(to_string(num));
        }

        struct Less {
            bool operator()(const string& a, const string& b) const {
                return a + b > b + a;
            }
        };

        sort(strs.begin(), strs.end(), Less{});

        string buffer;
        for (auto& s : strs) buffer += s;

        if (!buffer.empty() && buffer[0] == '0') return "0";
        return buffer;
    }
};